### PR TITLE
Add linter and error for code submissions that use global scope variables / methods

### PR DESCRIPTION
**Pull request by khoaguin**
[Issue](https://github.com/OpenMined/Heartbeat/issues/1282)

### CODE REVIEW

**Comment by khoaguin** [comment]
image

**Comment by khoaguin** [comment]
Hi @IrinaMBejan, currently the fix is like below: if the user put a `global` inside the code, it will display a warning. And if it's submitted to the server, then there will be an error showing the exception 
image

**Comment by khoaguin** [comment]
Hi @IrinaMBejan, currently the fix is like below
image

Also the way we do it now is to put the check in the `syft_function` decorator to run it locally first before submitting to the server

File: packages/syft/src/syft/service/code/user_code.py [hunk 1]
```diff
@@ -1285,26 +1294,48 @@ def generate_unique_func_name(context: TransformContext) -> TransformContext:
     return context
 
 
-def process_code(
-    context: TransformContext,
+def _check_global(code_tree: ast.Module) -> GlobalsVisitor | SyftWarning:
```
Comment (shubham3121) [inline packages/syft/src/syft/service/code/user_code.py]: we can remove `_` from function name, possibly rename to `check_for_global_vars` ?

File: packages/syft/src/syft/service/code/user_code.py [hunk 2]
```diff
@@ -1285,26 +1294,48 @@ def generate_unique_func_name(context: TransformContext) -> TransformContext:
...
+    try:
+        v.visit(code_tree)
+    except Exception:
+        raise SyftException(
+            "Your code contains (a) global variable(s), which is not allowed"
+        )
+    return v
+
+
+def _parse_code(raw_code: str) -> ast.Module | SyftWarning:
```
Comment (shubham3121) [inline packages/syft/src/syft/service/code/user_code.py]: Remove `_` from function name. Let's move `parse_code` and `check_for_global_vars` to `service/code/utils.py`

File: packages/syft/src/syft/service/code/user_code.py [hunk 3]
```diff
@@ -1329,6 +1360,25 @@ def process_code(
     return unparse(wrapper_function)
 
 
+def process_code(
```
Comment (shubham3121) [inline packages/syft/src/syft/service/code/user_code.py:1372]: 👍🏾

**Review by shubham3121 — CHANGES_REQUESTED** [review]
@khoaguin left a few minor comments, otherwise LGTM !! 💯

File: packages/syft/src/syft/service/code/user_code.py (hunk 2 above)
Comment (khoaguin) [reply packages/syft/src/syft/service/code/user_code.py]: Thanks for the review @shubham312. Done!

**Review by shubham3121 — APPROVED** [review]

**Pull request merged**